[FEATURE] WorkSessions functionality

ipz234vas opened the pull request:
- Implement CRUD operations with WorkSessions
- Implement DataLoader for WorkSessions
- Add WorkSession mutation and query
- Add types for WorkSession requests and responses
- Fix Permission enum

ipz234vas committed:
[FEATURE] WorkSessions functionality

- Implement CRUD operations with WorkSessions
- Implement DataLoader for WorkSessions
- Add WorkSession mutation and query
- Add types for WorkSession requests and responses
- Fix Permission enum

mmv-vt231 commented on timetracker.Server/API/WorkSession/WorkSessionMutation.cs:
```diff
@@ -0,0 +1,80 @@
+                .ResolveAsync(async context =>
+                {
+                    var inputSession = context.GetArgument<WorkSessionModel>("session");
+                    inputSession.SetBy = "system";
```
Add Enum for SetBy

mmv-vt231 replied:
Add Enum for SetBy

mmv-vt231 commented on timetracker.Server/API/WorkSession/WorkSessionMutation.cs:
```diff
@@ -0,0 +1,80 @@
+
+                    return await workSessionRepository.AddAsync(inputSession);
+                });
+            Field<WorkSessionResponseType>("stopSession")
```
Rename field

mmv-vt231 commented on timetracker.Server/API/WorkSession/WorkSessionMutation.cs:
```diff
@@ -0,0 +1,80 @@
+                    var inputSession = context.GetArgument<WorkSessionModel>("session");
+                    inputSession.SetBy = "manually";
+
+                    return await workSessionRepository.AddAsync(inputSession); ;
```
Delete extra semicon

mmv-vt231 commented on timetracker.Server/API/WorkSession/WorkSessionMutation.cs:
```diff
@@ -0,0 +1,80 @@
+                    return await workSessionRepository.UpdateAsync(session);
+                });
+            Field<WorkSessionResponseType>("EditSession")
+                .Authorize()
```
Add global Authorize to this class

mmv-vt231 commented on timetracker.Server/API/WorkSession/WorkSessionMutation.cs:
```diff
@@ -0,0 +1,80 @@
+
+                    return await workSessionRepository.AddAsync(inputSession); ;
+                });
+            Field<StringGraphType>("DeleteSession")
```
Return BooleanGraphType

mmv-vt231 commented on timetracker.Server/Domain/Entities/WorkSession.cs:
```diff
@@ -6,11 +6,12 @@ namespace timetracker.Server.Domain.Entities
-        public Guid? EditedBy { get; set; } 
-        public DateTime? EditedAt { get; set; } 
+        public Guid? EditedById { get; set; } 
+        public User? EditedBy { get; set; }
```
Remove EditedBy property

mmv-vt231 commented on timetracker.Server/Domain/Entities/WorkSession.cs:
```diff
@@ -6,11 +6,12 @@ namespace timetracker.Server.Domain.Entities
         public string SetBy { get; set; } 
-        public Guid? EditedBy { get; set; } 
-        public DateTime? EditedAt { get; set; } 
+        public Guid? EditedById { get; set; } 
```
Rename EditedById to EditorId

ipz234vas committed:
[FIX] WorkSessions functionality

- Fix WorkSession model
- Remove WorkSessionRepository
- Add SetBy enum
- Add GetLastWorkSession mutation for user
